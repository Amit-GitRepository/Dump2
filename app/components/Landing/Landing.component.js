import bannerImage from '../../assets/bannerArea.png';
import launchPageTestIDs from '../../config/testid/launchPage';
import noop from 'lodash/noop';
import OTPPrompt from '../BillUsage/OTPPrompt/OTPPrompt.component';
import Proptypes from 'prop-types';
import React, {Component} from 'react';
import styles from './Landing.style';
import {Banner, Button, Card, CircleShadowSeparator, IconTextInput, ISText, KeyboardScrollView, SquareIconButton} from '../Shared';
import {colors} from '../../themes/constants.styles';
import {Platform} from '../../utils/reactNative.util';
import {translate} from '../../language/i18n/helper';
import {View} from 'react-native';

class Landing extends Component {

  state = {
    showOTPModal: false,
    newMSISDN: this.props.msisdn,
    emptyMsisdn: false
  };

  componentWillReceiveProps (nextProps) {
    if (this.props.msisdn !== nextProps.msisdn) {
      this.setState({newMSISDN: nextProps.msisdn});
    }
    if (nextProps.validateOTPStatus === 'true') {
      this.props.onPayPress(this.state.newMSISDN);
    }
  }

   confirmOTP = (otp) => {
     this.props.validateOTP({msisdn: this.state.newMSISDN, otp: otp});
   };

   sendOTP = () => {
     const {newMSISDN} = this.state;
     if (this.validateMSISDN(newMSISDN)) {
       this.props.sendOTP(newMSISDN);
       this.setState({showOTPModal: true});
     }
   };

   validateMSISDN = (msisdn) => {
     this.setState({emptyMsisdn: !msisdn});
     return !!msisdn;
   }

   payPress = () => {
     const {newMSISDN} = this.state;
     if (this.validateMSISDN(newMSISDN)) {
       this.props.onPayPress(newMSISDN);
     }
   };

   setMSISDN = (msisdn) => this.setState({newMSISDN: msisdn, emptyMsisdn: !msisdn});;

   onClosePress = () => this.setState({showOTPModal: false});

   render () {
     const {onLogin, onRegister, msisdn, sendOTPStatus, validateOTPStatus, navigatePayOthers} = this.props;
     const {newMSISDN, showOTPModal, emptyMsisdn} = this.state;
     const inputTextPadding = Platform.OS === 'ios' ? styles.paddingIOS : styles.paddingAndroid;
     return (
       <KeyboardScrollView contentContainerStyle={styles.contentContainer}>
         {(sendOTPStatus && showOTPModal) ?
           <OTPPrompt msisdn={newMSISDN} onLogin={onLogin} onConfirm={this.confirmOTP}
             onClosePress={this.onClosePress} validateOTPStatus={validateOTPStatus} sendOTP={this.sendOTP}/> : null}
         <Banner source={bannerImage} />
         <Card headerStyles={styles.cardHeaderStyles} containerStyles={styles.cardContainer} header={translate('LANDING__EASY_PAYMENT_TITLE')} elevation={2}>
           <View style={styles.cardBody}>
             <View style={styles.upperBody}>
               <IconTextInput multiline={false} style={{...styles.numberInput, ...inputTextPadding}} containerStyle={styles.numberInputContainer}
                 defaultValue={msisdn} placeholder={translate('LANDING__EASY_PAYMENT_INPUT_PLACEHOLDER')}
                 onChangeText={this.setMSISDN} type={'SECONDARY'}/>
               <ISText type={'SEMIBOLD'} style={(emptyMsisdn || this.props.sendOTPStatus === false) ? styles.errorText : styles.hideErrorText}>{(emptyMsisdn) ? translate('LANDING__EASY_PAYMENT_INPUT_ERROR') : translate('LANDING__EASY_PAYMENT_INVALID_INPUT_ERROR')}</ISText>
               <Button textType={'SEMIBOLD'} style={styles.payButton} textStyle={styles.payButtonText} text={translate('LANDING__EASY_PAYMENT_BUTTON_TEXT')}
                 onPress={this.payPress} accessibilityLabel='Landing_LoginButton'/>
             </View>
             <CircleShadowSeparator text={translate('LANDING__OR_SEPARATOR')}/>
             <View style={styles.lowerBody} accessibilityLabel='Landing_check&buy&pay'>
               <View style={styles.boxButtonContainer}>
                 <SquareIconButton onPress={this.sendOTP} iconName='check-usage' iconColor={colors.SECONDARY_TMOVE} title={translate('LANDING__CHECK_USAGE_TITLE')} subtitle={translate('LANDING__CHECK_USAGE_SUBTITLE')}/>
                 <SquareIconButton iconName='buy-package' iconColor={colors.SECONDARY_TONLINE} title={translate('LANDING__BUY_EXTRA_TITLE')} subtitle={translate('LANDING__BUY_EXTRA_SUBTITLE')}/>
                 <SquareIconButton iconName='pay-other' iconColor={colors.SECONDARY_TVISION} onPress={navigatePayOthers} title={translate('LANDING__PAY_OTHERS_TITLE')} subtitle={translate('LANDING__PAY_OTHERS_SUBTITLE')}/>
               </View>
             </View>
           </View>
         </Card>
         <View style={styles.fullAccessText}><ISText>{translate('LANDING__FULL_ACCESS_SEPARATOR')}</ISText></View>
         <Button textStyle={styles.loginButtonText} style={styles.loginButton} text={translate('LANDING__LOGIN_TRUEID_BUTTON_TEXT')} type='secondary' onPress={onLogin} accessibilityLabel={launchPageTestIDs.BTN__LOGIN_TRUEID_BUTTON_TEXT} testID={launchPageTestIDs.BTN__LOGIN_TRUEID_BUTTON_TEXT}/>
         <View style={styles.footer} accessibilityLabel='Landing_registerLink'>
           <ISText style={styles.dontHaveTrueId}>{translate('LANDING__REGISTER_TITLE')}</ISText>
           <Button textType={'REGULAR'} textStyle={styles.registerBtnText} text={translate('LANDING__REGISTER_SUBTITLE')} type='inline' onPress={onRegister}/>
         </View>
       </KeyboardScrollView>
     );
   }
}

Landing.defaultProps = {
  onLogin: noop,
  onRegister: noop,
  onPayPress: noop,
  msisdn: null,
  sendOTPStatus: false,
  showPopup: noop,
  hidePopup: noop,
  navigatePayOthers: noop
};

Landing.propTypes = {
  onLogin: Proptypes.func,
  onRegister: Proptypes.func,
  onPayPress: Proptypes.func,
  msisdn: Proptypes.string,
  sendOTPStatus: Proptypes.bool,
  validateOTPStatus: Proptypes.bool,
  sendOTP: Proptypes.func,
  validateOTP: Proptypes.func,
  navigatePayOthers: Proptypes.func
};

export default Landing;
