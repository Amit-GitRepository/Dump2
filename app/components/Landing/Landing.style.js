import {colors, fonts} from '../../themes/constants.styles';
import {contentContainerStyle, errorTextStyle, shadowScattered} from '../../themes/application.styles';

export default {
  contentContainer: {
    backgroundColor: colors.PRIMARY_BG_TEXT_CONTRAST,
    ...contentContainerStyle
  },
  errorText: [errorTextStyle, {
    paddingVertical: 10,
    justifyContent: 'center',
    alignItems: 'center'
  }],
  hideErrorText: [errorTextStyle, {
    paddingBottom: 5,
    paddingVertical: 0,
    height: 16,
    color: colors.TRANSPARENT
  }],
  upperBody: {
    alignSelf: 'stretch',
    paddingHorizontal: 20,
    paddingTop: 0,
    paddingBottom: 0,
    flexGrow: 1,
    justifyContent: 'space-around'
  },
  lowerBody: {
    paddingTop: 5,
    paddingBottom: 9,
    paddingHorizontal: 7,
    alignSelf: 'stretch',
    backgroundColor: colors.PRIMARY_BG_SEPARATOR,
    flexGrow: 1,
    justifyContent: 'space-around'
  },
  boxButtonContainer: {
    alignSelf: 'stretch',
    flexDirection: 'row',
    justifyContent: 'space-around'
  },
  footer: {
    paddingVertical: 5,
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
    backgroundColor: colors.PRIMARY_BG_SEPARATOR
  },
  dontHaveTrueId: {
    color: colors.LANDING_TITLE_BG
  },
  title: {
    padding: 7,
    fontSize: fonts.FONT_SIZE_MEDIUM,
    textAlign: 'center',
    color: colors.PRIMARY_BG_TEXT_CONTRAST,
    backgroundColor: colors.LANDING_TITLE_BG
  },
  numberInputContainer: {
    borderRadius: 8
  },
  numberInput: {
    textAlign: 'center'
  },
  payButtonText: {
    fontSize: fonts.FONT_SIZE_MEDIUM,
    paddingVertical: 4
  },
  payButton: {
    paddingVertical: 3
  },
  loginButton: {
    paddingVertical: 5,
    marginHorizontal: 30,
    marginBottom: 16,
    borderColor: colors.PRIMARY_ACTIONABLE,
    borderWidth: 1
  },
  loginButtonText: {
    fontSize: fonts.FONT_SIZE_MEDIUM,
    color: colors.PRIMARY_ACTIONABLE
  },
  registerBtnText: {
    fontSize: fonts.FONT_SIZE_SMALL
  },
  cardContainer: {
    marginHorizontal: 10,
    marginTop: -45,
    borderRadius: 8,
    ...shadowScattered
  },
  cardBody: {
    paddingTop: 21,
    marginTop: -15,
    borderWidth: 1,
    borderBottomLeftRadius: 8,
    borderBottomRightRadius: 8,
    borderColor: colors.FRENCH_GREY
  },
  fullAccessText: {
    alignItems: 'center',
    paddingBottom: 8,
    paddingTop: 10
  },
  cardHeaderStyles: {
    padding: 8
  },
  paddingIOS: {
    paddingVertical: 10
  },
  paddingAndroid: {
    paddingVertical: 5
  }
};
