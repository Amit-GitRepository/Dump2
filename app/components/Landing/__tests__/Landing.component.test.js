import Landing from '../Landing.component';
import React from 'react';
import ReactSixteenAdapter from 'enzyme/build/adapters/ReactSixteenAdapter';
import renderer from 'react-test-renderer';
import {configure, shallow} from 'enzyme';

configure({adapter: new ReactSixteenAdapter()});

describe('Landing component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Landing selectedLanguage='th'/>);
    const component = renderer.create(<Landing selectedLanguage='th'/>).toJSON();
    expect(component).toMatchSnapshot();
    expect(wrapper).toBeDefined();
  });
});
