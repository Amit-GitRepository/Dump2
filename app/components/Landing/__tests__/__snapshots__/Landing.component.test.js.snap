// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing component renders correctly 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Object {
      "backgroundColor": "#FFFFFF",
      "flexDirection": "column",
      "flexGrow": 1,
      "justifyContent": "space-between",
    }
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutoAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  onScroll={[Function]}
  overScrollMode="auto"
  scrollEventThrottle={0}
  showsVerticalScrollIndicator={true}
>
  <View>
    <View
      style={Object {}}
    >
      <Image
        getSize={[Function]}
        isBackground={false}
        onLayout={[Function]}
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../app/assets/bannerArea.png",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      elevation={2}
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 8,
            "flex": 1,
            "marginHorizontal": -1,
            "overflow": "hidden",
            "shadowColor": "#303C46",
            "shadowOffset": Object {
              "height": 2,
              "width": 1,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 1,
          },
          Object {
            "borderRadius": 8,
            "marginHorizontal": 10,
            "marginTop": -45,
            "shadowColor": "#303C46",
            "shadowOffset": Object {
              "height": 2,
              "width": 1,
            },
            "shadowOpacity": 0.5,
            "shadowRadius": 2,
          },
        ]
      }
    >
      <View
        onPress={null}
        style={
          Array [
            Object {
              "zIndex": 1,
            },
            Object {},
          ]
        }
      >
        <View>
          <View
            style={
              Object {
                "borderWidth": 0,
                "elevation": 3,
                "shadowColor": "#303C46",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 1,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderBottomColor": "#F93E60",
                    "paddingHorizontal": 5,
                  },
                  Object {
                    "borderBottomColor": "#303C46",
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#E9E9EA",
                        "padding": 10,
                      },
                      Object {
                        "backgroundColor": "#303C46",
                        "marginHorizontal": -5,
                        "padding": 8,
                      },
                    ],
                    Object {},
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#303C46",
                      },
                      Array [
                        Object {
                          "color": "#000000",
                          "fontSize": 18,
                        },
                        Object {
                          "color": "#FFFFFF",
                        },
                      ],
                      Object {
                        "fontFamily": "SukhumvitSet-Bold",
                      },
                    ]
                  }
                >
                  Easy Payment & Top Up
                </Text>
              </View>
            </View>
          </View>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#303C46",
                  "fontSize": 24,
                },
                Array [
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "bottom": 9.6,
                    "marginBottom": -9.6,
                  },
                  Object {},
                ],
                Object {
                  "fontFamily": "iservice",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            Ó§Å
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderColor": "#CFCFD1",
            "borderWidth": 1,
            "marginTop": -15,
            "paddingTop": 21,
          }
        }
      >
        <View
          style={
            Object {
              "alignSelf": "stretch",
              "flexGrow": 1,
              "justifyContent": "space-around",
              "paddingBottom": 0,
              "paddingHorizontal": 20,
              "paddingTop": 0,
            }
          }
        >
          <View
            accessibilityLabel={undefined}
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#605E71",
                  "borderRadius": 4,
                  "borderWidth": 2,
                  "flexDirection": "row",
                },
                Object {
                  "borderColor": "#B9C1C9",
                  "borderWidth": 1,
                },
                Object {
                  "borderRadius": 8,
                },
              ]
            }
            testID={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 35,
                  },
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "transparent",
                    "color": "#B9C1C9",
                    "padding": 5,
                  },
                  Object {
                    "fontFamily": "iservice",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
            <TextInput
              allowFontScaling={true}
              defaultValue={null}
              multiline={false}
              onChangeText={[Function]}
              placeholder="Enter Mobile / Service number"
              placeholderTextColor="#B9C1C9"
              style={
                Array [
                  Object {
                    "alignSelf": "stretch",
                    "borderRadius": 4,
                    "flex": 1,
                    "fontSize": 16,
                    "paddingHorizontal": 0,
                    "paddingVertical": 10,
                  },
                  Object {
                    "paddingVertical": 10,
                    "textAlign": "center",
                  },
                ]
              }
              underlineColorAndroid="transparent"
            />
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#303C46",
                },
                Array [
                  Object {
                    "color": "#F93E60",
                    "fontSize": 14,
                    "paddingBottom": 10,
                    "paddingVertical": 5,
                    "textAlign": "center",
                  },
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "paddingVertical": 10,
                  },
                ],
                Object {
                  "fontFamily": "SukhumvitSet-SemiBold",
                },
              ]
            }
          >
            Invalid TrueMove H / Service Number
          </Text>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel="Landing_LoginButton"
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "#F93E60",
                "borderRadius": 50,
                "opacity": 1,
                "padding": 10,
                "paddingVertical": 3,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "stretch",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#303C46",
                      },
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "fontSize": 16,
                        },
                        Object {
                          "fontSize": 18,
                          "paddingVertical": 4,
                        },
                      ],
                      Object {
                        "fontFamily": "SukhumvitSet-SemiBold",
                      },
                    ]
                  }
                >
                  Pay Now
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignSelf": "stretch",
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "elevation": 1,
                "flex": 1,
                "padding": 10,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#E9E9EA",
                "flex": 1,
                "padding": 10,
              }
            }
          />
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "#FFFFFF",
                "borderBottomLeftRadius": 50,
                "borderBottomRightRadius": 50,
                "paddingBottom": 4,
                "paddingHorizontal": 10,
                "position": "absolute",
                "top": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#605E71",
                  "fontFamily": "SukhumvitSet-Bold",
                  "fontSize": 14,
                }
              }
            >
              OR
            </Text>
          </View>
        </View>
        <View
          accessibilityLabel="Landing_check&buy&pay"
          style={
            Object {
              "alignSelf": "stretch",
              "backgroundColor": "#E9E9EA",
              "flexGrow": 1,
              "justifyContent": "space-around",
              "paddingBottom": 9,
              "paddingHorizontal": 7,
              "paddingTop": 5,
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "stretch",
                "flexDirection": "row",
                "justifyContent": "space-around",
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "borderRadius": 15,
                  "flex": 1,
                  "opacity": 1,
                  "paddingHorizontal": 5,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#FFFFFF",
                      "borderRadius": 8.5,
                      "elevation": 3,
                      "marginBottom": 4,
                      "paddingBottom": 2,
                      "paddingTop": 6,
                      "shadowColor": "#303C46",
                      "shadowOffset": Object {
                        "height": 2,
                        "width": 1,
                      },
                      "shadowOpacity": 0.5,
                      "shadowRadius": 2,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        Array [
                          Object {
                            "color": "#F93E60",
                            "fontSize": 30,
                            "padding": 1,
                          },
                          Object {
                            "color": "#FBBC1E",
                          },
                        ],
                        Object {
                          "fontFamily": "iservice",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    Ó§Ö
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#000000",
                        "fontFamily": "SukhumvitSet-Bold",
                        "fontSize": 16,
                      }
                    }
                  >
                    Check
                  </Text>
                  <View
                    style={
                      Object {
                        "alignItems": "flex-end",
                        "flexDirection": "row-reverse",
                        "marginTop": -5,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      noOfLines={1}
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "color": "#000000",
                          "fontFamily": "SukhumvitSet-Text",
                          "fontSize": 14,
                        }
                      }
                    >
                      Usage
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "borderRadius": 15,
                  "flex": 1,
                  "opacity": 1,
                  "paddingHorizontal": 5,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#FFFFFF",
                      "borderRadius": 8.5,
                      "elevation": 3,
                      "marginBottom": 4,
                      "paddingBottom": 2,
                      "paddingTop": 6,
                      "shadowColor": "#303C46",
                      "shadowOffset": Object {
                        "height": 2,
                        "width": 1,
                      },
                      "shadowOpacity": 0.5,
                      "shadowRadius": 2,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        Array [
                          Object {
                            "color": "#F93E60",
                            "fontSize": 30,
                            "padding": 1,
                          },
                          Object {
                            "color": "#009FE3",
                          },
                        ],
                        Object {
                          "fontFamily": "iservice",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    Ó§É
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#000000",
                        "fontFamily": "SukhumvitSet-Bold",
                        "fontSize": 16,
                      }
                    }
                  >
                    Buy
                  </Text>
                  <View
                    style={
                      Object {
                        "alignItems": "flex-end",
                        "flexDirection": "row-reverse",
                        "marginTop": -5,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      noOfLines={1}
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "color": "#000000",
                          "fontFamily": "SukhumvitSet-Text",
                          "fontSize": 14,
                        }
                      }
                    >
                      Extra Package
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "borderRadius": 15,
                  "flex": 1,
                  "opacity": 1,
                  "paddingHorizontal": 5,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#FFFFFF",
                      "borderRadius": 8.5,
                      "elevation": 3,
                      "marginBottom": 4,
                      "paddingBottom": 2,
                      "paddingTop": 6,
                      "shadowColor": "#303C46",
                      "shadowOffset": Object {
                        "height": 2,
                        "width": 1,
                      },
                      "shadowOpacity": 0.5,
                      "shadowRadius": 2,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 12,
                        },
                        Array [
                          Object {
                            "color": "#F93E60",
                            "fontSize": 30,
                            "padding": 1,
                          },
                          Object {
                            "color": "#E82E8A",
                          },
                        ],
                        Object {
                          "fontFamily": "iservice",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    Ó§Ñ
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#000000",
                        "fontFamily": "SukhumvitSet-Bold",
                        "fontSize": 16,
                      }
                    }
                  >
                    Pay
                  </Text>
                  <View
                    style={
                      Object {
                        "alignItems": "flex-end",
                        "flexDirection": "row-reverse",
                        "marginTop": -5,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      noOfLines={1}
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "color": "#000000",
                          "fontFamily": "SukhumvitSet-Text",
                          "fontSize": 14,
                        }
                      }
                    >
                      for Other
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "paddingBottom": 8,
          "paddingTop": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#303C46",
            "fontFamily": "SukhumvitSet-Text",
          }
        }
      >
        Get Full Access
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel="landingPage_button_loginWithTrueId"
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#F93E60",
          "borderRadius": 50,
          "borderWidth": 1,
          "marginBottom": 16,
          "marginHorizontal": 30,
          "opacity": 1,
          "padding": 10,
          "paddingVertical": 5,
        }
      }
      testID="landingPage_button_loginWithTrueId"
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#303C46",
                },
                Array [
                  Object {
                    "color": "#303C46",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#F93E60",
                    "fontSize": 18,
                  },
                ],
                Object {
                  "fontFamily": "SukhumvitSet-Bold",
                },
              ]
            }
          >
            Login with TrueID
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityLabel="Landing_registerLink"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E9E9EA",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingVertical": 5,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#605E71",
            "fontFamily": "SukhumvitSet-Text",
          }
        }
      >
        Don't have TrueID ?
      </Text>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "transparent",
            "borderRadius": 50,
            "borderWidth": 0,
            "opacity": 1,
            "padding": 5,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#303C46",
                  },
                  Array [
                    Object {
                      "color": "#F93E60",
                      "fontSize": 16,
                    },
                    Object {
                      "fontSize": 14,
                    },
                  ],
                  Object {
                    "fontFamily": "SukhumvitSet-Text",
                  },
                ]
              }
            >
              Register Here
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
